eZ/Publish/Core/FieldType/RelationList/Type.php

```diff
@@ -110,7 +120,134 @@ public function validateFieldSettings($fieldSettings)
                         );
                     }
                     break;
+                case 'selectionLimit':
+                    if (!is_int($value)) {
+                        $validationErrors[] = new ValidationError(
+                            "Setting '%setting%' value must be of integer type",
+                            null,
+                            array(
+                                '%setting%' => $name,
+                            ),
+                            "[$name]"
+                        );
+                    }
+                    if ($value < 0) {
+                        $validationErrors[] = new ValidationError(
+                            "Setting '%setting%' value cannot be lower than 0",
+                            null,
+                            array(
+                                '%setting%' => $name,
+                            ),
+                            "[$name]"
+                        );
+                    }
+                    break;
+            }
+        }
+
+        return $validationErrors;
+    }
+
+    /**
+     * Validates the validatorConfiguration of a FieldDefinitionCreateStruct or FieldDefinitionUpdateStruct.
+     *
+     * @param mixed $validatorConfiguration
+     *
+     * @return \eZ\Publish\SPI\FieldType\ValidationError[]
+     */
+    public function validateValidatorConfiguration($validatorConfiguration)
+    {
+        $validationErrors = array();
+
+        foreach ($validatorConfiguration as $validatorIdentifier => $constraints) {
+            if ($validatorIdentifier !== 'RelationListValueValidator') {
+                $validationErrors[] = new ValidationError(
+                    "Validator '%validator%' is unknown",
+                    null,
+                    array(
+                        '%validator%' => $validatorIdentifier,
+                    ),
+                    "[$validatorIdentifier]"
+                );
+
+                continue;
             }
+
+            foreach ($constraints as $name => $value) {
+                switch ($name) {
+                    case 'selectionLimit':
```

Comment:
In this case `if ($name === 'selectionLimit') {` would reduce number of nested blocks. If there's need for more constraints, then it should be redesigned in a form of some sort of dispatcher to avoid spaghetti code.